### PR TITLE
feat(azure): adding AzureCliCredential as a fallback authentication option

### PR DESCRIPTION
**Pull request by ali42**
Uses AzureCliCredential to add a Bearer token for the Azure OpenAI api calls. 
This is useful for running promptfoo locally without keys (as mentioned in https://github.com/promptfoo/promptfoo/issues/2099) 

TODO:
- [x] Update auth error messages so they guide the user better on their auth options
- [x] Update the Docs

Additional changes included in this PR:
- Client Secret Credential Token was not passed as an Authorization Bearer token in the http calls so fixes that too. 
- `_cachedApiKey` approach was not sufficient for the token as with each call this would mean getting a new token. So instead we find out the auth headers (api or token) at the constructor stage and cache that instead.

Suggested Future PR Changes:
- enable token based auth for OpenAI Assistants as well

### CODE REVIEW

File: src/providers/azure.ts [hunk 1]
```diff
@@ -228,60 +234,101 @@ export class AzureGenericProvider implements ApiProvider {
...
+      getEnvString('AZURE_API_KEY') ||
+      this.env?.AZURE_OPENAI_API_KEY ||
+      getEnvString('AZURE_OPENAI_API_KEY')
+    );
+  }
+
+  getApiKeyOrThrow(): string {
+    const apiKey = this.getApiKey();
+    if (!apiKey) {
+      throw new Error('Azure OpenAI API key must be set');
```
Comment (mwiemer-microsoft) [inline src/providers/azure.ts]: Maybe we can link to some docs in this error message?

**Review by mwiemer-microsoft — COMMENTED** [review]
Looks great to me! To clarify, I work for Microsoft, but not Azure, and am mostly commenting as someone else who needs this feature for work :)

**Comment by typpo** [comment]
Huge thanks @ali42!  This looks great and I'll get it in the next release.

**Comment by ali42** [comment]
Thank you both! Tested the 0.97.0 locally as well with azure cli auth, looks great!

**Pull request merged**